File: implementations/java/src/main/java/com/axisrobo/aep/EventTypes.java
package com.axisrobo.aep;

import com.axisrobo.harmovela.context.ContextMemoryTypes;
import com.axisrobo.harmovela.delegation.DelegationTypes;
import com.axisrobo.harmovela.recovery.RecoveryTypes;
import com.axisrobo.harmovela.state.StateTypes;
import com.axisrobo.harmovela.tool.ToolTypes;
import java.util.Set;
import java.util.HashSet;

public final class EventTypes {
    private EventTypes() {}

    private static final Set<String> LEGACY_TYPES;

    static {
        Set<String> types = new HashSet<>(Set.of(
            "event.redelivered", "event.replayed", "event.dead_lettered",
            "task.submitted", "task.accepted", "task.started", "task.blocked", "task.progress",
            "task.output", "task.completed", "task.failed", "task.cancel.requested", "task.cancelled", "task.timed_out",
            "agent.message.sent", "agent.message.received", "agent.message.failed",
            "agent.request.created", "agent.response.created", "agent.decision.recorded",
            "environment.observed", "environment.changed", "environment.alerted", "environment.error"
        ));
        types.addAll(ContextMemoryTypes.EVENT_TYPES);
        types.addAll(DelegationTypes.EVENT_TYPES);
        types.addAll(RecoveryTypes.EVENT_TYPES);
        types.addAll(StateTypes.EVENT_TYPES);
        types.addAll(ToolTypes.EVENT_TYPES);
        LEGACY_TYPES = Set.copyOf(types);
    }

    public static boolean isStandardEventType(String type) {
        return type != null && (com.axisrobo.harmovela.event.registry.EventTypes.isStandardEventType(type)
            || LEGACY_TYPES.contains(type));
    }
}

File: implementations/go/aep/event_types.go
package aep

import (
	"github.com/axisrobo/harmovela/context"
	"github.com/axisrobo/harmovela/event"
)

var legacyStandardEventTypes = func() map[string]bool {
	m := map[string]bool{
		"event.redelivered":              true,
		"event.replayed":                 true,
		"event.dead_lettered":            true,
		"tool.call.requested":            true,
		"tool.call.accepted":             true,
		"tool.call.rejected":             true,
		"tool.call.started":              true,
		"tool.call.progress":             true,
		"tool.call.output":               true,
		"tool.call.completed":            true,
		"tool.call.failed":               true,
		"tool.call.cancel.requested":     true,
		"tool.call.cancelled":            true,
		"tool.call.timed_out":            true,
		"task.submitted":                 true,
		"task.accepted":                  true,
		"task.started":                   true,
		"task.blocked":                   true,
		"task.progress":                  true,
		"task.output":                    true,
		"task.completed":                 true,
		"task.failed":                    true,
		"task.cancel.requested":          true,
		"task.cancelled":                 true,
		"task.timed_out":                 true,
		"agent.message.sent":             true,
		"agent.message.received":         true,
		"agent.message.failed":           true,
		"agent.request.created":          true,
		"agent.response.created":         true,
		"agent.decision.recorded":        true,
		"environment.observed":           true,
		"environment.changed":            true,
		"environment.alerted":            true,
		"environment.error":              true,
		"belief.revised":                 true,
		"belief.conflict.detected":       true,
		"freshness.expired":              true,
		"freshness.window.changed":       true,
		"delegation.requested":           true,
		"delegation.accepted":            true,
		"delegation.rejected":            true,
		"delegation.handoff.completed":   true,
		"delegation.escalated":           true,
		"interruption.requested":         true,
		"interruption.acknowledged":      true,
		"interruption.saved":             true,
		"interruption.resumed":           true,
		"interruption.cancelled":         true,
		"compensation.requested":         true,
		"compensation.completed":         true,
		"provenance.attestation.added":   true,
		"provenance.attestation.revoked": true,
		"provenance.chain.truncated":     true,
		"state.snapshot.requested":       true,
		"state.snapshot.ready":           true,
		"state.delta.applied":            true,
		"state.invalidated":              true,
	}
	for k, v := range context.EventTypes {
		m[k] = v
	}
	return m
}()

// IsStandardEventType reports whether a type is supported by the legacy aep adapter.
func IsStandardEventType(typ string) bool {
	return event.IsStandardEventType(typ) || legacyStandardEventTypes[typ]
}
